### PR TITLE
Use threads local sums of C_h and I_h to reduce memory consumption 

### PR DESCRIPTION
**Pull request by Tortar**
This should shave away the biggest part of allocations with big models without changing anything in terms of how the model works

### CODE REVIEW

File: src/markets/search_and_matching.jl [hunk 1]
```diff
@@ -256,8 +257,8 @@ function initialize_variables_retail_market(firms, rotw, prop, agg, w_act, w_ina
     Q_d_i_g = zeros(size(firms.Y_i)..., G)
     Q_d_m_g = zeros(size(rotw.Y_m)..., G)
 
-    C_h_g = zeros(H, G)
-    I_h_g = zeros(H, G)
+    C_h_t = zeros(H, multi_threading ? Threads.nthreads() : 1)
+    I_h_t = zeros(H, multi_threading ? Threads.nthreads() : 1)
```
Comment (AldoGl) [inline src/markets/search_and_matching.jl:261]: nice!

File: test/deterministic/one_run_deterministic.jl [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
     end
 
+    model, data = run_deterministic(parameters, initial_conditions, T, false)
+    model2, data2 = run_deterministic(parameters, initial_conditions, T, false)
+    model3, data3 = run_deterministic(parameters, initial_conditions, T, true)
+    
     # loop over the data fields and compare them
     for field in fieldnames(typeof(data))
         @test isapprox(getproperty(data, field), getproperty(data2, field), rtol = 0.001)
+        @test isapprox(getproperty(data2, field), getproperty(data3, field), rtol = 0.001)
```
Comment (AldoGl) [inline test/deterministic/one_run_deterministic.jl:24]: @Tortar probably we can get to lower tolerances here can't we? (0.00001 should also work?)

**Review by AldoGl — APPROVED** [review]
This looks great! I just give a couple of suggestions but we can merge for me

File: src/markets/search_and_matching.jl [hunk 3]
```diff
@@ -538,8 +540,8 @@ function perform_retail_market!(
     @~ Q_d_i_g[:, g] .= @view(S_f[1:I]) .- @view(S_fg[1:I])
     @~ Q_d_m_g[:, g] .= @view(S_f[(I + 1):end]) .- @view(S_fg[(I + 1):end])
 
-    @~ C_h_g[:, g] .= b
-    @~ I_h_g[:, g] .= d
+    @~ C_h_t[:, i] .+= b
```
[the remark this answers is not in the document]
Comment (Tortar) [reply src/markets/search_and_matching.jl:543]: yes, it's just an adjustments for using n_threads < n_sectors

File: src/markets/search_and_matching.jl (hunk 1 above)
Comment (Tortar) [reply src/markets/search_and_matching.jl:261]: thanks :)

File: test/deterministic/one_run_deterministic.jl (hunk 2 above)
Comment (Tortar) [reply test/deterministic/one_run_deterministic.jl:24]: try before merging

Comment (Tortar) [reply test/deterministic/one_run_deterministic.jl]: added 2 0s

**Pull request merged**